add: ciの追加

Mikuto0831 opened the pull request:
# なぜやるのか
- formatやlintを行うことにより、コードの一貫性を持たせるため
- githubでのチーム開発初心者たちに、「PRつくったのにmergeできないなんで！？」な体験をしてもらうため

# 何をしたか
- prettier による format の機能を追加
- eslint による lint の機能を追加
- actions によるCIの追加

# やらなかったこと
- lintをしようしたCI
  - lintルールの策定に時間がかかると判断したため、別PRに切り分ける

# 見てほしい点
- lintルール

Mikuto0831 committed:
add:ciの追加

Mikuto0831 committed:
fmt:フォーマットの実施

Mikuto0831 committed:
Merge origin/main with theirs strategy

Mikuto0831 committed:
update: typescript

Mikuto0831 committed:
fmt:フォーマットを実施

Mikuto0831 committed:
add:formatコマンドのショートカットfmtを作成

Mikuto0831 committed:
fix:lintで発生していたエラーをpackage.jsonでtypeをmoduleに指定することにより解決

R31K4G3 commented on package.json:
```diff
@@ -7,15 +7,21 @@
+    "lint:check": "eslint \"src/**/*.ts\"",
+    "typecheck": "tsc --noEmit"
   },
+  "type": "module",
```
type を module にしてしまうと Dockerfile に書かれている `CMD [ "node", "./dist/main.js" ]` が動かなくなるのでは？

Mikuto0831 replied:
単純に把握してなかったです
moduleをやめてもlintが通ることを確認したので消しました

R31K4G3 commented:
.eslintrc.json と eslint.config.js があるけど，これって両方必要？

R31K4G3 commented on eslint.config.js:
```diff
@@ -0,0 +1,51 @@
+import eslint from '@eslint/js';
```
package.json の devDependencies に @eslint/js って 追加しなくていいの？

Mikuto0831 replied:
追加しました

Mikuto0831 commented:
> .eslintrc.json と eslint.config.js があるけど，これって両方必要？

.eslintrc.json消し忘れです

Mikuto0831 committed:
fix: PRレビューコメントの対応

- package.jsonのtype: moduleを削除（CommonJS出力のため不要）
- @eslint/jsをdevDependenciesに追加
- 古い.eslintrc.jsonを削除
- eslint.config.jsをCommonJS形式に変更

Mikuto0831 committed:
add: CIにDockerビルド・起動テストを追加

- docker compose buildの成功を確認
- コンテナが正常に起動しDiscordログインまで到達することを確認
- TokenInvalidエラーが出ることで、コード自体は正常に動作していることを検証

Mikuto0831 committed:
fix: CIのDockerテストを修正

- docker compose logsではなく、upの出力を直接ファイルに保存
- set +eでエラーを一時的に無視し、終了コードを取得
- ログを表示してデバッグしやすくする
- grepの結果で成功/失敗を明示的に判定

Mikuto0831 commented:
docker compose upを行い、コードが実行されるところまで確認するCIを追加しました

R31K4G3 commented on eslint.config.js:
```diff
@@ -0,0 +1,51 @@
+                setInterval: 'readonly',
+                clearInterval: 'readonly',
+                NodeJS: 'readonly',
+            },
```
https://www.npmjs.com/package/globals を使ったほうがいいかも。

Mikuto0831 replied:
あーこういうのがあるんだ
これにしてみる

R31K4G3 commented:
TypeScript ESLint ，recommended よりも recommendedTypeChecked のほうが良いと思う。

Mikuto0831 committed:
fix: globalsパッケージを使用してNode.jsグローバル変数を定義

- globalsパッケージをdevDependenciesに追加
- eslint.config.jsでglobals.nodeを使用
- 手動で定義していたNode.jsグローバル変数をglobalsパッケージに置き換え
- NodeJSは個別に追加（globals.nodeに含まれていないため）

Mikuto0831 committed:
fix: recommended-type-checkedに変更

- tseslint.configs.recommendedからflat/recommended-type-checkedに変更
- より厳格な型チェックを有効化
- no-unused-varsを無効化してTypeScript ESLint版を優先

Mikuto0831 commented:
やってみたー

R31K4G3 commented on .prettierrc.json:
```diff
@@ -0,0 +1,9 @@
+    "printWidth": 80,
+    "tabWidth": 2,
+    "useTabs": false,
+    "endOfLine": "lf"
```
```suggestion
    "endOfLine": "lf",
    "arrowParens": "avoid"
```

R31K4G3 replied:
`(x) => { ... }` よりも `x => { ... }` と書いている箇所が多そうなので，この変更を提案。

Mikuto0831 replied:
採用

R31K4G3 commented on .prettierrc.json:
```diff
@@ -0,0 +1,9 @@
+    "semi": true,
+    "trailingComma": "es5",
+    "singleQuote": false,
+    "printWidth": 80,
```
printWidth は 80 でいいのかどうか。

Mikuto0831 replied:
別PRで修正

Mikuto0831 replied:
ちげぇ、これformatだやる

Mikuto0831 replied:
120のほうが見た感じすきだったから、120にしようかな
defaultは80らしい

Mikuto0831 replied:
考えの参考
https://github.com/prettier/prettier/issues/12424

ノートだとちょっと長いか？まぁ長くなりすぎないようにすればよし

R31K4G3 commented on eslint.config.js:
ESLint のコンフィグの書き方を，もうちょっと丁寧に調べたほうがいいかもしれないので，あとで調べる。
現在の eslint.config.js ，ESLint が想定された動きをするのか，どうかわからない。

Mikuto0831 replied:
dousiyo

Mikuto0831 replied:
Lintを別PRにした！

Mikuto0831 committed:
delete: ciからlintを除く

Mikuto0831 committed:
Merge remote-tracking branch 'origin/main' into add/ci-workflow

Mikuto0831 committed:
change: formatの設定を変更

Mikuto0831 committed:
fmt: フォーマットの実施

R31K4G3 commented on eslint.config.js:
別PRにするなら，このPRではこのファイル消しませんか

R31K4G3 commented on package.json:
```diff
@@ -7,15 +7,19 @@
   "license": "ISC",
-"devDependencies": {
+  "devDependencies": {
+    "@eslint/js": "^9.4.0",
```
別PRにするなら devDependencies の更新はしなくていいんじゃないすか

Mikuto0831 committed:
delete:eslint

R31K4G3 commented:
`"@eslint/js": "^9.4.0"` だけじゃなくて `"globals": "^16.5.0"` もじゃない？

Mikuto0831 commented:
まじか

R31K4G3 commented on package.json:
```diff
@@ -24,6 +27,7 @@
     "eslint-plugin-import": "^2.29.1",
     "eslint-plugin-n": "^17.6.0",
     "eslint-plugin-promise": "^7.2.1",
+    "globals": "^16.5.0",
```
​
